Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java

```diff
@@ -654,4 +661,29 @@ private static String getRelatesTo(AssertionType assertion) {
     private static String getRelatesToValue(List<String> relatesTo) {
         return StringUtils.isNotEmpty(relatesTo.get(0)) ? relatesTo.get(0) : null;
     }
+
+    private String getRequestorUserId(NhinTargetSystemType target, UserType userInfo) {
+        if (userInfo == null || StringUtils.isBlank(userInfo.getUserName()) || !checkDistinguishedName(userInfo.
+            getUserName().trim())) {
+            String alias = ExchangeManagerHelper.getExchangeAlias(target.getExchangeName());
+            try {
+                X509Certificate cert = getCertificateManager().getCertificateBy(alias);
+                return cert.getSubjectX500Principal().getName();
+            } catch (CertificateManagerException ex) {
+                LOG.error("Unable to load certificate with alias {}", alias, ex);
+                return null;
+            }
+        }
+        return userInfo.getUserName().trim();
+    }
+
+    private static boolean checkDistinguishedName(String userName) {
```

Comment:
We have this exact method already in HOKSAMLAssertionBuilder. Could we extract this to a shared utility class instead of duplicating code here?